src/EnergyPlus/OutputProcessor.cc

```diff
@@ -4028,358 +3086,50 @@ namespace OutputProcessor {
         auto &op = state.dataOutputProcessor;
 
         if (varType == VariableType::Integer) {
-            *op->IVariableTypes(keyVarIndex).VarPtr.Which = SetIntVal;
+            OutVarInt *varInt = dynamic_cast<OutVarInt *>(op->outVars[keyVarIndex]);
+            assert(varInt != nullptr);
+            *varInt->Which = SetIntVal;
         } else if (varType == VariableType::Real) {
-            *op->RVariableTypes(keyVarIndex).VarPtr.Which = SetRealVal;
+            OutVarReal *varReal = dynamic_cast<OutVarReal *>(op->outVars[keyVarIndex]);
+            assert(varReal != nullptr);
+            *varReal->Which = SetRealVal;
         } else if (varType == VariableType::Meter) {
-            op->EnergyMeters(keyVarIndex).CurTSValue = SetRealVal;
+            op->meters[keyVarIndex]->CurTSValue = SetRealVal;
         }
-    }
-
-    // returns the string corresponding to the OutputProcessor::Unit enum in brackets
-    std::string unitEnumToStringBrackets(EnergyPlus::OutputProcessor::Unit const unitIn)
-    {
-        // J.Glazer - August/September 2017
-        return " [" + unitEnumToString(unitIn) + "]";
-    }
+    } // SetInternalVariableValue()
 
     // returns the unit string for a DDVariableTypes item and custom string when customEMS is used
-    std::string unitStringFromDDitem(EnergyPlusData &state, int const ddItemPtr // index provided for DDVariableTypes
+    std::string unitStringFromDDitem(EnergyPlusData &state, int const ddNum // index provided for DDVariableTypes
     )
     {
-        // J.Glazer - August/September 2017
-        OutputProcessor::Unit ddUnit = state.dataOutputProcessor->DDVariableTypes(ddItemPtr).units;
-        if (ddUnit != OutputProcessor::Unit::customEMS) {
-            return unitEnumToStringBrackets(ddUnit);
-        } else {
-            return " [" + state.dataOutputProcessor->DDVariableTypes(ddItemPtr).unitNameCustomEMS + "]";
-        }
-    }
-
-    // returns the string corresponding to the OutputProcessor::Unit enum
```

Comment:
Thse are constexpr arrays of string_view now.

Reply:
This has been needed for a while. Thanks.